lib/buffer.js

```diff
@@ -1259,7 +1259,28 @@ function atob(input) {
   if (arguments.length === 0) {
     throw new ERR_MISSING_ARGS('input');
   }
-  input = `${input}`;
+
+  if (input === undefined || input === false || typeof input === 'number') {
+    throw lazyDOMException(
+      'The string to be decoded is not correctly encoded.',
+      'ValidationError');
```

Comment:
I'm not sure whether `ValidationError` is appropriate here. Same with line 1281. Let me know what you think.

Reply:
I don't think this test is necessary, we should simply try to convert the argument to a string, the error can be thrown later.

Reply:
Addressed!